### ISSUE TITLE
Reflection

### ISSUE THREAD

**Issue by emmaxshen** [e-shen2022/emma_blog#46]
- [x] Skills you have obtained
- Built a **base in basics of programming** (data structures, algorithms)
- Developing a mindset for **problem-solving and logical reasoning** (AP College Board practice, experience in creating, reading, and analyzing code) 
- Proficiency in **Python** (low-level programs for simple game purposes to machine learning algorithms w/user interface) 
- Experience in frontend + **integrating frontend & backend** (HTML, CSS, Javascript!!)
- **Team collaboration** and leadership skills (Scrum master all three trimesters, communication among team members and with teacher) 

- [x] My CSP Journey 

1. <b>Tri 1</b>: Setting up the environment, learning new basics through lessons & hacks, familiarizing myself with team coding, lots of frontend work to make websites more visually appealing (HTML, CSS) & some python. N@M project is a website that allows a user to check off their symptoms and we produce an output of their most likely sickness. 
2. <b>Tri 2</b>: Starting working with making coding LESSONS and how to effectively teach classmates. Also got exposed to many other new concepts from other people's presentations. Started working towards integrating frontend & backend together by fetching backend data and displaying it on the website. Created a website that takes users' skin type and their desired product and outputs a personal product recommendation. Studied for the AP CSP exam where got to develop solid logical reasoning and analytical skills 
3. <b>Tri 3</b>: Began diving more in-depth with my projects and opening myself up to exploring new unknown territories in Computer Science. Created real-time human facial recognition software that translates subject to an emoji using a trained convolutional neural network. Really put in A LOT of work in learning how to go through the machine learning cycle of training data, creating the model, training the model, integrating the front end, and continuing to train the model for higher accuracy. Lots of trial and error and nights dedicated to debugging. Learned the concept behind using OpenCV and allowing the OS system to access the firmware. Figured out how to sink two functions using events to stop or switch threads. Very fun!

- [x] Highs and Lows 

<b>Highs</b>: 

- Tri 3 N@M on **machine learning** (really got to dive deep into the unknown and interesting project)
- Tri 2 **Lesson presentation** where my group presented about random() and applications in real-life and algorithms (Was fun to create a curriculum and learn how to effectively teach peers)

<b>Lows</b>: 

- Tri 3 N@M also. I had to do it by myself and felt really **lost in errors and new concepts I'd never understood**
- Tri 2 didn't really feel like I grew THAT MUCH in my technical experience. Still very unclear on how to connect frontend and backend

- [x] Future plans with CompSci

- When I take CSA next year, I plan to **prioritize cultivating my intellectual curiosity** and taking on projects or doing hacks related to my interests. Particularly, I want to find more datasets on Kaggle that I want to analyze and test different machine-learning algorithms on it to find which one is the most accurate. I want to develop an understanding of when to use what algorithm given a problem.
- I hope to **continue as the Scrum Master** to facilitate collaboration in my team and communication with the teacher. If not, I would also like to venture into being a DevOps since I haven't really worked on deployment before
- Over the summer I'm going to **take an online Java course** and hope to build a base throughout the next school year! 
- Some skills I want to refine: **integrating frontend and backend (javascript)**, finding easy ways to deploy personal projects

- [x] Future plans in Education/Career

- Ideally, I want my career choice to reflect **a combination of my current strengths, and skills I want to cultivate in the future**. My current strengths are being meticulous, organized, and presenting myself to others. I love STEM subjects such as biology and math, and also found myself finding Computer Science concepts and projects very interesting. Although I know I do not have strong technical expertise or maybe a good enough logical mind, I want to try my best to pursue the things I'm not good at. Thus, for now a future job position that appeals to me is one that allows the intersection of computational science and business. I'm attending a summer research program at UCSB this summer where I will spend a six weeks working 1 on 1 with a professor on ongoing, interdisciplinary research. I hope to pick a project that aligns with my interests. I want to learn if maybe I want to pursue academic career as a researcher, or industry job in tech, or even serving role such as product manager. 
- **I am interested in the following majors**: Data Science, Information Systems, Computational Biology, Computer Science
